shelley-ma/formal-spec/value.tex

```diff
@@ -1,125 +1,289 @@
+parts of the system as a multi-asset representation which is more suitable in particular
+usecases.
+
+\subsection{$\ValMonoid$}
+
+Figure \ref{fig:ValMonoid} gives names and types of functions which
+make up the $\ValMonoid$ structure. Note that we use overloaded notation
+for pointwise partial order comparisons of $\ValMonoid$ terms elsewhere in the spec. That is,
+we use
+
+\[ =,~<,~>,~\leq,~\geq \]
+
+in place of the following functions for partial ordering, respectively
+
+\[ \fun{pointwise}~(=),~\fun{pointwise}~(<),~\fun{pointwise}~(>),~\fun{pointwise}~(\leq),~\fun{pointwise}~(\geq) \]
+
+%%
+%% Figure ValMonoid and its Functions
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \mathsf{ValType} ~\in~ \mathsf{Type}  \\
+      & \text{A type}
+      \nextdef
+      %
+      & \fun{zero} ~\in~ \mathsf{ValType} \\
+      & \text{Additive identity of $\mathsf{ValType}$}
+      \nextdef
+      %
+      & (+) ~\in~ \mathsf{ValType} \to \mathsf{ValType} \to \mathsf{ValType}\\
+      & \text{Addition on $\mathsf{ValType}$}
+      \nextdef
+      %
+      & (*) ~\in~ \mathsf{Integer} \to \mathsf{ValType} \to \mathsf{ValType}\\
+      & \text{Scale $\mathsf{ValType}$ value by integral constant}
+      \nextdef
+      %
+      & \fun{coin} ~\in~ \mathsf{ValType} \to \Coin\\
+      & \text{Return the Ada contained in a $\mathsf{ValType}$ term}
+      \nextdef
+      %
+      & \fun{inject} ~\in~ \Coin \to \mathsf{ValType}\\
+      & \text{Create a $\mathsf{ValType}$ term containing only this amount of Ada}
+      \nextdef
+      %
+      & \fun{adaOnly} ~\in~ \mathsf{ValType} \to \Bool \\
+      & \fun{adaOnly} =
+        \begin{cases}
+          \True & \fun{inject} \circ \fun{coin}~ = ~\fun{id}_{\mathsf{ValType}} \\
+          \False & \text{otherwise}
+        \end{cases} \\
+      & \text{Check if a given $\mathsf{ValType}$ term contains only Ada}
+      \nextdef
+      %
+      & \fun{size} ~\in~ \mathsf{ValType} \to \MemoryEstimate\\
+      & \text{Return the size in words of a $\mathsf{ValType}$ term}
+      \nextdef
+      %
+      & \fun{pointwise} ~\in~ (\Integer \to \Integer \to \Bool) \to (\mathsf{ValType} \to \mathsf{ValType} \to \Bool)\\
+      & \text{Pointwise comparison of two $\mathsf{ValType}$ terms}
+  \end{align*}
+  \caption{The $\ValMonoid$ Structure}
+  \label{fig:ValMonoid}
+\end{figure}
 
-When multi-asset support on the ledger is introduced, Ada is still expected to be
-the most common type of token on the ledger.
-The $\Coin$ type is used to represent an amount of Ada.
+\subsection{$\Coin$ in the Allegra era}
+
+For the Allegra era, we require the $\ValMonoid$ asset representation to support
+only a single asset, Ada, represented via $\Coin$. This section defines the
+$\ValMonoid$ functions for the $\Coin$ type, see Figure
+\ref{fig:coin}. Addition and multiplication are defined in the usual way
+(as on $\Integer$).
+
+When multi-asset support on the ledger is introduced, Ada ($\Coin$) is still expected to be
```

Comment:
I'm not sure what the purpose of this sentence is. Also, most common by what metric?

Reply:
by the metric that every single output contains some

Reply:
If this is the intended meaning of this sentence, I think 'expected' is not the correct word, since this is enforced by the ledger. Maybe 'will' instead?